test/test_custom_ops.py

```diff
@@ -1842,6 +1843,43 @@ def test_impl_device_invalid(self):
         with self.assertRaisesRegex(RuntimeError, "Expected one of cpu, cuda"):
             torch.library.impl("blah::blah", "somethingsomething")
 
+    def test_autograd_function_backed_op(self):
+        cpp_source = """
+struct CustomOpAutogradFunction : public torch::autograd::Function<CustomOpAutogradFunction> {
+  static constexpr bool is_traceable = true;
+
+  static torch::Tensor forward(
+      torch::autograd::AutogradContext* ctx,
+      torch::Tensor x) {
+    return x;
+  }
+
+  static torch::autograd::variable_list backward(
+      torch::autograd::AutogradContext *ctx,
+      torch::autograd::variable_list grad_output) {
+    return grad_output;
+  }
+};
+
+torch::Tensor custom_op_with_autograd(torch::Tensor x) {
+  return CustomOpAutogradFunction::apply(x);
+}
```

Comment:
This is not exactly what you want: this is a c++ autograd::Function that is bound to Python, not a custom op. To make it a custom op we need some TORCH_LIBRARY macros:
```
TORCH_LIBRARY(mylib, m) {
    m.define(...);
    m.impl(...);
}
```

Reply:
added the registration, but i'm not sure if the test is invoking the op or the function directly. I get some not declared in scope errors when I try to name the impl function differently than the python one